client/chain-spec/src/lib.rs

```diff
@@ -149,6 +149,8 @@ pub trait ChainSpec: BuildStorage + Send {
 	fn add_boot_node(&mut self, addr: MultiaddrWithPeerId);
 	/// Return spec as JSON.
 	fn as_json(&self, raw: bool) -> Result<String, String>;
+	/// Return spec as JSON value.
+	fn as_json_value(&self, raw: bool) -> Result<serde_json::Value, String>;
```

Comment:
Can we not change the function above?  I don't see any reason why we need this two times. And yes, I see that one is a String and the other one a value.

Reply:
Making this change causes `build_spec_works` to fail, because of `Error: Service(Other("Error generating spec json: u128 is not supported"))`. I think this is because `serde_json::to_value` panics on `u128`s while `serde_json::to_string_pretty` does not seem to.

This is a gotcha that we need to look out for in the RPC call too.

Reply:
Where is the `u128` coming from? 

We already use this "hack" in transaction payment: https://github.com/paritytech/substrate/blob/master/frame/transaction-payment/rpc/runtime-api/src/lib.rs#L40-L43

Reply:
I'm not sure. This only happens when running `substrate build-spec --dev`, but not without `--dev`. The balance type for the node is `u128`, so that could be why?

Reply:
Yeah sounds right, but we still should not have 2 functions which are doing the same. If you need a serde::Value for the RPC, you could still deserialize the String to a `Value`. I know this is also shit, but we should not blow up our code base because Serde isn't able to handle this correctly...